client/homebrew/pages/vaultPage/vaultPage.jsx

```diff
@@ -0,0 +1,478 @@
+		//Different searches use the update or total props to make only the necessary queries and functions
+
+		if (!validateForm()) {
+			return;
+		}
+
+		setSearching(true);
+		setError(null);
+
+		const performSearch = async ({ title, author, count, v3, legacy }) => {
+			updateUrl(title, author, count, v3, legacy, page);
+			console.log(title, author, count, v3, legacy);
+			if ((title || author) && (v3 || legacy)) {
+				try {
+					const response = await request.get(
+						`/api/vault?title=${title}&author=${author}&v3=${v3}&legacy=${legacy}&count=${count}&page=${page}`
+					);
+					if (response.ok) {
+						updateStateWithBrews(response.body.brews, page);
+					} else {
+						throw new Error(`Error: ${response.status}`);
+					}
+				} catch (error) {
+					console.log('error at loadPage: ', error);
+					setError(
+						`${
+							error.response
+								? error.response.status
+								: error.message
+						}`
+					);
+					updateStateWithBrews([], 1);
+				}
+			} else {
+				setError('404');
+			}
+		};
+
+		const loadTotal = async ({ title, v3, legacy }) => {
+			setTotalBrews(null);
+			setError(null);
+			if ((title || author) && (v3 || legacy)) {
+				try {
+					const response = await request.get(
+						`/api/vault/total?title=${title}&author=${author}&v3=${v3}&legacy=${legacy}`
+					);
+
+					if (response.ok) {
+						setTotalBrews(response.body.totalBrews);
+					} else {
+						throw new Error(
+							`Failed to load total brews: ${response.statusText}`
+						);
+					}
+				} catch (error) {
+					console.log('error at loadTotal: ', error);
+					setError(`${error.response.status}`);
+					updateStateWithBrews([], 1);
+				}
+			}
+		};
+
+		const title = titleRef.current.value || '';
+		const author = authorRef.current.value || '';
+		const count = countRef.current.value || 10;
+		const v3 = v3Ref.current.checked != false;
+		const legacy = legacyRef.current.checked != false;
+
+		console.log(title);
+		if (update) {
+			setTitle(title);
+			setAuthor(author);
+			setCount(count);
+			setV3(v3);
+			setLegacy(legacy);
+		}
+
+		// Perform search with the latest input values, because state is not fast enough
+		performSearch({ title, author, count, v3, legacy });
+
+		if (total) {
+			loadTotal({ title, author, v3, legacy });
+		}
+	};
+
+	const renderNavItems = () => (
+		<Navbar>
+			<Nav.section>
+				<Nav.item className="brewTitle">
+					Vault: Search for brews
+				</Nav.item>
+			</Nav.section>
+			<Nav.section>
+				<NewBrew />
+				<HelpNavItem />
+				<RecentNavItem />
+				<Account />
+			</Nav.section>
+		</Navbar>
+	);
+
+	const validateForm = () => {
+		//form validity: title or author must be written, and at least one renderer set
+		const { current: titleInput } = titleRef;
+		const { current: legacyCheckbox } = legacyRef;
+		const { current: v3Checkbox } = v3Ref;
+		const { current: authorInput } = authorRef;
+
+		const isTitleValid = titleInput.validity.valid && titleInput.value;
```

Comment:
Why not just
```suggestion
		const isTitleValid = titleRef.current.validity.valid && titleInput.value;
```

on all of these?